cruise-control/src/main/scala/com/linkedin/kafka/cruisecontrol/executor/ExecutorUtils.scala

```diff
@@ -88,13 +90,29 @@ object ExecutorUtils {
                 if (!currentReplicaAssignment.contains(destinationBroker)) {
                   if (currentReplicaAssignment.contains(sourceBroker)) {
                     // this is a normal movement.
-                    if (destinationBroker != null)
-                      (currentReplicaAssignment :+ destinationBroker.toInt).filter(_ != sourceBroker) 
-                    else
+                    if (destinationBroker != null) {
```

Comment:
It seems clearer to have the if statements on the balancing action. So we have something like:
```
if (balacning action is replicamovement) {
  getNewReplicasForReplicaMovement
} else if (balancing action is replica addition) {
   getNewReplicasForReplicaAddition
} else if (balancing action is replica deletion) {
   getNewReplicasForReplicaDeletion
} else {
   exception
}```